src/Utilities/StdHelpers.hpp

```diff
@@ -283,3 +284,86 @@ std::string formatted_string(const std::string& fmt, Args... args) {
   }
   return rtn;
 }
+
+// Arithmetic operators for std::array<T, Dim>
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim>& operator+=(
+    std::array<T, Dim>& lhs,
+    const std::array<T, Dim>& rhs) noexcept(noexcept(lhs[0] += rhs[0])) {
+  for (size_t i = 0; i < Dim; ++i) {
+    gsl::at(lhs, i) += gsl::at(rhs, i);
+  }
+  return lhs;
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator+(
+    const std::array<T, Dim>& lhs,
+    const std::array<T, Dim>&
+        rhs) noexcept(noexcept(std::declval<std::array<T, Dim>&>() += rhs) and
+                      noexcept(std::array<T, Dim>{lhs})) {
+  std::array<T, Dim> result = lhs;
+  result += rhs;
+  return result;
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim>& operator-=(
+    std::array<T, Dim>& lhs,
+    const std::array<T, Dim>& rhs) noexcept(noexcept(lhs[0] -= rhs[0])) {
+  for (size_t i = 0; i < Dim; ++i) {
+    gsl::at(lhs, i) -= gsl::at(rhs, i);
+  }
+  return lhs;
+}
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator-(
+    const std::array<T, Dim>& lhs,
+    const std::array<T, Dim>&
+        rhs) noexcept(noexcept(std::declval<std::array<T, Dim>&>() -= rhs) and
+                      noexcept(std::array<T, Dim>{lhs})) {
+  std::array<T, Dim> result = lhs;
+  result -= rhs;
+  return result;
+}
+
+template <size_t Dim, typename T, typename U>
+inline std::array<T, Dim> operator*(const std::array<T, Dim>& lhs,
+                                    const U& scale) noexcept(noexcept(lhs[0] *
+                                                                      scale)) {
+  std::array<T, Dim> result{};
+  for (size_t i = 0; i < Dim; ++i) {
+    gsl::at(result, i) = gsl::at(lhs, i) * scale;
+  }
+  return result;
+}
+
+template <size_t Dim, typename T, typename U>
+inline std::array<T, Dim> operator*(
+    const U& scale,
+    const std::array<T, Dim>& rhs) noexcept(noexcept(rhs* scale)) {
+  return rhs * scale;
+}
+
+template <size_t Dim, typename T, typename U>
+inline std::array<T, Dim> operator/(const std::array<T, Dim>& lhs,
+                                    const U& scale) noexcept(noexcept(lhs[0] /
+                                                                      scale)) {
+  std::array<T, Dim> result{};
```

Comment:
I just noticed, the `noexcept`ness of these also requires `noexcept(std::array<T, Dim>{})` be true